### ISSUE TITLE
Sidebar constantly refreshes itself (v3.8.2)

### ISSUE THREAD

**Issue by Arthaey** [piroor/treestyletab#2936]
## Short description

Since yesterday, my TST sidebar started refreshing itself constantly. I noticed that there was also a new release right around that time. So I downgraded back to 3.8.1 and the problem stopped.

(I apologize for not being able to test with a clean Firefox profile right now, but I figured it would be better to at least let you know about the issue than to say nothing about the latest release just because of that.)

## Steps to reproduce

 1. Install TST **3.8.2**.
 2. Load several tabs, then change focus between them (as though browsing).

## Expected result

The TST sidebar should show my tabs.

## Actual result

The TST sidebar refreshes every couple seconds. It becomes blank, then shows my tabs, then becomes blank again, in an infinite cycle.

## Workaround

Downgrade to TST **3.8.1**.

## Environment

 * Platform (OS): Mac OS 11.3 (Big Sur)
 * Version of Firefox: Firefox Developer Edition, 89.0b10
 * Version (or revision) of Tree Style Tab: 3.8.2

**Comment by gotyaoi** [comment]
Have you tried toggling the "Optimize tree restoration with cache" setting off and on in the TST options? That can sometimes help.

**Comment by I60R** [comment]
I have the same issue after the same update, except refresh happens only after focus on other tab. Toggling "optimize tree restoration with cache" doesn't help

**Comment by piroor** [comment]
I couldn't reproduce this on Firefox 89.0 on my macOS 11.4 Big Sur. Hmm...

Could you try the latest development build? https://github.com/piroor/treestyletab#development-builds

**Comment by I60R** [comment]
Continues to refresh for me on development build, at least with my existing tree structure.

BTW, Manjaro Linux, Firefox 89, sway, large number of tabs (probably > 1000)

**Comment by piroor** [comment]
@I60R Thanks! Does the problem happen on a clean profile? If not, could you bisect minimum conditions like as I described at #2931 ?

**Comment by I60R** [comment]
So, I've found the cause of problem: [VivaldiFox extension](https://addons.mozilla.org/en-US/firefox/addon/vivaldifox/) — TST don't refreshes after I disabled it. Still cannot reproduce the same behavior on a clean profile with both extensions installed.

**Comment by irvinm** [comment]
@I60R can you close this item?  You can always reopen if there really is a problem.

**Comment by piroor** [comment]
@I60R Thanks, I got what happens. The reloading by tab switch is triggered due to the option "Themes" => "Frame" => "Apply page colors" in VivaldiFox Options. Due to the change 89defeadad2fd34a74bb868e96172d203c9ec0c9 introduced for another issue #2903, TST now updates the URL of the sidebar panel including "bgcolor" information to apply theme color ASAP. I didn't imagine cases like this: changing background color too frequently. Hmm, the change looks needed to be reverted...

**Comment by piroor** [comment]
I've removed codes changing the sidebar panel too frequently with the commit 2f5a486.

**Comment by I60R** [comment]
@piroor Thank you, I installed development build and confirm that this fixes the problem. I think this issue could be closed

**Comment by irvinm** [comment]
@piroor I think this can be closed.
